### PR TITLE
Feat/DWS-418, DWS-344 - add bucket unique check

### PR DESCRIPTION
**Pull request by rybnovdws**

### CODE REVIEW

**Comment by bl1t3** [comment]
API Should return a valid and s3 interface compatible error

File: pkg/minio/api-handlers.go [hunk 1]
```diff
@@ -448,7 +450,7 @@ func (h objectAPIHandlersWrapper) PutBucketHandler(w http.ResponseWriter, r *htt
 func (h objectAPIHandlersWrapper) HeadBucketHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	defer mon.Task()(&ctx)(nil)
-	if err := h.bucketPrefixSubstitution(w, r); err != nil {
+	if err := h.bucketPrefixSubstitution(w, r, "HeadObject"); err != nil {
```
Comment (bl1t3) [inline pkg/minio/api-handlers.go]: Just want to make sure that 3rd argument shouldn't be "HeadBucket", should it?

File: pkg/minio/api-handlers.go [hunk 2]
```diff
@@ -498,7 +500,7 @@ func (h objectAPIHandlersWrapper) DeleteBucketEncryptionHandler(w http.ResponseW
 func (h objectAPIHandlersWrapper) DeleteBucketHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	defer mon.Task()(&ctx)(nil)
-	if err := h.bucketPrefixSubstitution(w, r); err != nil {
+	if err := h.bucketPrefixSubstitution(w, r, "DeleteObject"); err != nil {
```
Comment (bl1t3) [inline pkg/minio/api-handlers.go]: Same as a comment above

Comment (rybnovdws) [reply pkg/minio/api-handlers.go]: fixed

File: pkg/minio/api-handlers.go (hunk 1 above)
Comment (rybnovdws) [reply pkg/minio/api-handlers.go]: fixed

**Review by bl1t3 — APPROVED** [review]

**Pull request merged**